### PR TITLE
[ONNX 1] Add Backend section to the left page of Tutorials layout page

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
This PR adds a `Backends` section to the left menu which initially will have a placeholder for the series of ONNX export tutorials based on Torch Dynamo backend
    
A table of content tree page will have static URL `https://pytorch.org/tutorials/beginner/onnx/intro_onnx.html` that will be referenced on PyTorch user document for `torch.onnx module`

Following is a screenshot of the resulting page

Preview page: https://docs-preview.pytorch.org/pytorch/tutorials/2550/beginner/onnx/intro_onnx.html

Fixes: https://github.com/pytorch/tutorials/issues/2552

### CODE REVIEW

**Comment by thiagocrepaldi** [comment]
https://github.com/pytorch/tutorials/pull/2541/ will be added to this page as soon as it can be merged

**Comment by thiagocrepaldi** [comment]
@msaroufim FYI

**Review by msaroufim — CHANGES_REQUESTED** [review]
I'm generally OK with having a `tutorials/onnx` but having onnx be a in the left hand navigation is not something I'm as excited about since I'd then have to also be OK with other compiler vendors doing the same

Is there any way we can roll this content in to the deployment tab?

**Comment by thiagocrepaldi** [comment]
> I'm generally OK with having a `tutorials/onnx` index page but having onnx be a in the left hand navigation is not something I'm as excited about since I'd then have to also be OK with other compiler vendors doing the same
> 
> Is there any way we can roll this content in to the deployment tab?

If we could have submenus under `Deploying PyTorch Models in Production` menu, that probably would help already, but I think this is not a trivial task by quickly going through the code. That is why we proposed a ONNX menu on its own.

We do understand the duality of ONNX, especially after pytorch 2.x series. On one hand, `torch.onnx` is within pytorch project under the github.com/pytorch umbrella, like torchvision, torchtext, and torchaudio. Each of these other pytorch components have a menu on their own. From this point of view, ONNX module as pytorch's component could have a menu on its own. On the other hand, after pt 2.0 the concept of dynamo backends got generalized and vendors started implementing theirs and adding all of them might be tricky. 

Also, because we introduced a new ONNX exporter based on Dynamo (`torch.onnx.dynamo_export`) but we still support the older one based on TorchScript (`torch.onnx.export`), we plan in having tutorials helping on the decision-making process of which one to use, which wouldn't be a "deployment" tutorial per se, but very related. Other related but not exactly deployment might be 1) migration process from the TorchScript to the Dynamo based exporter and 2) working around limitations of the exporter, helping users to fix their model for successful export. There is a whole bunch of ONNX-related tutorials to cover lack of documentation of the previous exporter that we plan that may not fit directly into the "deployment" umbrella and having them sprinkled in different menus would be a bit messy

I hope that helps making our case :)

**Comment by thiagocrepaldi** [comment]
Changed the left menu to `Backends` instead of `ONNX` as per offline discussion with @svekars

**Comment by msaroufim** [comment]
Ping me when the doc renders I still don't see the most recent changes

**Review by msaroufim — APPROVED** [review]

**Comment by msaroufim** [comment]
ok just stamped thanks!

**Comment by svekars** [comment]
Thanks! I will do an editorial pass. We will merge all 2.1 tutorials on the day of the release.

File: beginner_source/onnx/intro_onnx.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+format for representing machine learning models. The ``torch.onnx`` module provides APIs to
+capture the computation graph from a native PyTorch :class:`torch.nn.Module` model and convert
+it into an `ONNX graph <https://github.com/onnx/onnx/blob/main/docs/IR.md>`_.
+
+The exported model can be consumed by any of the many
+`runtimes that support ONNX <https://onnx.ai/supported-tools.html#deployModel>`_,
+including Microsoft's `ONNX Runtime <https://www.onnxruntime.ai>`_.
+
+.. Note::
+    Currently there are two flavors of ONNX exporter APIs,
```
Comment (svekars) [inline beginner_source/onnx/intro_onnx.py:20]:
```suggestion
    Currently, there are two flavors of ONNX exporter APIs,
```

**Review by svekars — APPROVED** [review]
LGTM, just a couple of nits.

**Comment by thiagocrepaldi** [comment]
> LGTM, just a couple of nits.

nits fixed

File: .jenkins/validate_tutorials_built.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 NOT_RUN = [
     "beginner_source/basics/intro",  # no code
+    "beginner_source/onnx/intro_onnx",
```
Comment (svekars) [inline .jenkins/validate_tutorials_built.py:13]:
```suggestion
    "beginner_source/onnx/intro_onnx", # has no usable code and serves as a landing for ONNX tutorials
```

**Review by svekars — APPROVED** [review]

**Pull request merged**